### PR TITLE
Clang format 2

### PR DESCRIPTION
**Pull request by mlknz**

### CODE REVIEW

File: Source/Engine/Private/Window.cpp [hunk 1]
```diff
@@ -13,54 +13,48 @@ namespace Details
...
         const auto callback = [](GLFWwindow*, int32_t key, int32_t, int32_t action, int32_t mods)
-            {
-                const KeyInput keyInput{
-                    static_cast<Key>(key),
-                    static_cast<KeyAction>(action),
-                    ModifierFlags(static_cast<uint32_t>(mods))
-                };
+        {
+            const KeyInput keyInput{ static_cast<Key>(key), static_cast<KeyAction>(action),
+                ModifierFlags(static_cast<uint32_t>(mods)) };
```
Comment (ExtinctionHD) [inline Source/Engine/Private/Window.cpp]: comma

File: Source/Engine/Private/Window.cpp [hunk 2]
```diff
@@ -13,54 +13,48 @@ namespace Details
...
         const auto callback = [](GLFWwindow*, int32_t button, int32_t action, int32_t mods)
-            {
-                const MouseInput mouseInput{
-                    static_cast<MouseButton>(button),
-                    static_cast<MouseButtonAction>(action),
-                    ModifierFlags(static_cast<uint32_t>(mods))
-                };
+        {
+            const MouseInput mouseInput{ static_cast<MouseButton>(button), static_cast<MouseButtonAction>(action),
+                ModifierFlags(static_cast<uint32_t>(mods)) };
```
Comment (ExtinctionHD) [inline Source/Engine/Private/Window.cpp]: comma

File: Source/Engine/Render/Private/OcclusionRenderer.cpp [hunk 3]
```diff
@@ -20,30 +20,31 @@ namespace Details
...
-                const ImageLayoutTransition layoutTransition{
-                    vk::ImageLayout::eUndefined,
-                    vk::ImageLayout::eDepthStencilAttachmentOptimal,
-                    PipelineBarrier::kEmpty
-                };
+        VulkanContext::device->ExecuteOneTimeCommands(
+                [&texture](vk::CommandBuffer commandBuffer)
+                {
+                    const ImageLayoutTransition layoutTransition{ vk::ImageLayout::eUndefined,
+                        vk::ImageLayout::eDepthStencilAttachmentOptimal, PipelineBarrier::kEmpty };
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Private/OcclusionRenderer.cpp]: comma

File: Source/Engine/Render/Private/OcclusionRenderer.cpp [hunk 4]
```diff
@@ -105,15 +100,10 @@ namespace Details
...
-                    vk::ShaderStageFlagBits::eVertex,
-                    defines),
+                    Filepath("~/Shaders/Hybrid/GBuffer.vert"), vk::ShaderStageFlagBits::eVertex, defines),
         };
 
-        const VertexInput vertexInput{
-            { vk::Format::eR32G32B32Sfloat },
-            0, vk::VertexInputRate::eVertex
-        };
+        const VertexInput vertexInput{ { vk::Format::eR32G32B32Sfloat }, 0, vk::VertexInputRate::eVertex };
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Private/OcclusionRenderer.cpp]: comma

File: Source/Engine/Render/Private/PathTracingRenderer.cpp [hunk 5]
```diff
@@ -239,17 +234,10 @@ void PathTracingRenderer::Render(vk::CommandBuffer commandBuffer, uint32_t image
...
-        const ImageLayoutTransition layoutTransition{
-            vk::ImageLayout::eUndefined,
-            vk::ImageLayout::eGeneral,
-            PipelineBarrier{
-                SyncScope::kWaitForNone,
-                SyncScope::kRayTracingShaderWrite
-            }
-        };
+        const ImageLayoutTransition layoutTransition{ vk::ImageLayout::eUndefined, vk::ImageLayout::eGeneral,
+            PipelineBarrier{ SyncScope::kWaitForNone, SyncScope::kRayTracingShaderWrite } };
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Private/PathTracingRenderer.cpp]: comma

File: Source/Engine/Render/Private/PathTracingRenderer.cpp [hunk 6]
```diff
@@ -272,24 +260,18 @@ void PathTracingRenderer::Render(vk::CommandBuffer commandBuffer, uint32_t image
...
-        const ImageLayoutTransition layoutTransition{
-            vk::ImageLayout::eGeneral,
+        const ImageLayoutTransition layoutTransition{ vk::ImageLayout::eGeneral,
             vk::ImageLayout::eColorAttachmentOptimal,
-            PipelineBarrier{
-                SyncScope::kRayTracingShaderWrite,
-                SyncScope::kColorAttachmentWrite
-            }
-        };
+            PipelineBarrier{ SyncScope::kRayTracingShaderWrite, SyncScope::kColorAttachmentWrite } };
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Private/PathTracingRenderer.cpp]: two commas

File: Source/Engine/Render/Stages/Private/GBufferStage.cpp [hunk 7]
```diff
@@ -72,58 +72,52 @@ namespace Details
...
-                const ImageLayoutTransition colorLayoutTransition{
-                    vk::ImageLayout::eUndefined,
-                    vk::ImageLayout::eGeneral,
-                    PipelineBarrier::kEmpty
-                };
+        VulkanContext::device->ExecuteOneTimeCommands(
+                [&renderTargets](vk::CommandBuffer commandBuffer)
+                {
+                    const ImageLayoutTransition colorLayoutTransition{ vk::ImageLayout::eUndefined,
+                        vk::ImageLayout::eGeneral, PipelineBarrier::kEmpty };
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Stages/Private/GBufferStage.cpp:95]: comma

File: Source/Engine/Render/Stages/Private/GBufferStage.cpp [hunk 8]
```diff
@@ -72,58 +72,52 @@ namespace Details
...
 
-                const ImageLayoutTransition depthLayoutTransition{
-                    vk::ImageLayout::eUndefined,
-                    vk::ImageLayout::eDepthStencilAttachmentOptimal,
-                    PipelineBarrier::kEmpty
-                };
+                    const ImageLayoutTransition depthLayoutTransition{ vk::ImageLayout::eUndefined,
+                        vk::ImageLayout::eDepthStencilAttachmentOptimal, PipelineBarrier::kEmpty };
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Stages/Private/GBufferStage.cpp:98]: comma

File: Source/Engine/Render/Stages/Private/GBufferStage.cpp [hunk 9]
```diff
@@ -149,22 +143,21 @@ namespace Details
...
         };
 
         const vk::CullModeFlagBits cullMode = materialFlags & MaterialFlagBits::eDoubleSided
-                ? vk::CullModeFlagBits::eNone : vk::CullModeFlagBits::eBack;
+                ? vk::CullModeFlagBits::eNone
+                : vk::CullModeFlagBits::eBack;
 
         const std::vector<BlendMode> blendModes(GBufferStage::kColorAttachmentCount, BlendMode::eDisabled);
 
         const std::vector<vk::PushConstantRange> pushConstantRanges{
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Stages/Private/GBufferStage.cpp:157]: can be removed

File: Source/Engine/Render/Stages/Private/ForwardStage.cpp [hunk 10]
```diff
@@ -118,20 +121,19 @@ namespace Details
...
-                    vk::ShaderStageFlagBits::eFragment, defines)
+                    Filepath("~/Shaders/Hybrid/Forward.frag"), vk::ShaderStageFlagBits::eFragment, defines),
         };
 
         const vk::CullModeFlagBits cullMode = materialFlags & MaterialFlagBits::eDoubleSided
-                ? vk::CullModeFlagBits::eNone : vk::CullModeFlagBits::eBack;
+                ? vk::CullModeFlagBits::eNone
+                : vk::CullModeFlagBits::eBack;
 
         const std::vector<vk::PushConstantRange> pushConstantRanges{
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Stages/Private/ForwardStage.cpp:133]: can be removed

File: Source/Engine/Render/Vulkan/Pipelines/Private/ComputePipeline.cpp [hunk 11]
```diff
@@ -1,35 +1,34 @@
...
 #include "Engine/Render/Vulkan/Shaders/ShaderHelpers.hpp"
+#include "Engine/Render/Vulkan/VulkanContext.hpp"
 
 std::unique_ptr<ComputePipeline> ComputePipeline::Create(const ShaderModule& shaderModule)
 {
     const std::vector<ShaderModule> shaderModules{ shaderModule };
 
-    const vk::PipelineShaderStageCreateInfo shaderStageCreateInfo
-            = ShaderHelpers::CreateShaderStagesCreateInfo(shaderModules).front();
+    const vk::PipelineShaderStageCreateInfo shaderStageCreateInfo =
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Vulkan/Pipelines/Private/ComputePipeline.cpp:10]: maybe it's possible to break before operators

File: Source/Engine/Render/Vulkan/Pipelines/Private/RayTracingPipeline.cpp [hunk 12]
```diff
@@ -106,13 +103,8 @@ namespace Details
...
 
-        return ShaderBindingTable{
-            CreateShaderGroupsBuffer(pipeline, groupCount),
-            offsets.at(ShaderGroupType::eRaygen),
-            offsets.at(ShaderGroupType::eMiss),
-            offsets.at(ShaderGroupType::eHit),
-            baseAlignment
-        };
+        return ShaderBindingTable{ CreateShaderGroupsBuffer(pipeline, groupCount), offsets.at(ShaderGroupType::eRaygen),
+            offsets.at(ShaderGroupType::eMiss), offsets.at(ShaderGroupType::eHit), baseAlignment };
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Vulkan/Pipelines/Private/RayTracingPipeline.cpp:107]: comma

File: Source/Engine/Render/Vulkan/Private/Device.cpp [hunk 13]
```diff
@@ -137,19 +134,17 @@ namespace Details
...
-            const Queues::Description& queuesDescription)
+    static std::vector<vk::DeviceQueueCreateInfo> CreateQueuesCreateInfo(const Queues::Description& queuesDescription)
     {
         static constexpr float queuePriority = 0.0;
 
-        std::vector<vk::DeviceQueueCreateInfo> queuesCreateInfo{
-            vk::DeviceQueueCreateInfo({}, queuesDescription.graphicsFamilyIndex, 1, &queuePriority)
-        };
+        std::vector<vk::DeviceQueueCreateInfo> queuesCreateInfo{ vk::DeviceQueueCreateInfo(
+                {}, queuesDescription.graphicsFamilyIndex, 1, &queuePriority) };
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Vulkan/Private/Device.cpp:142]: comma

File: Source/Engine/Render/Vulkan/Private/Device.cpp [hunk 14]
```diff
@@ -190,48 +185,40 @@ namespace Details
...
+                physicalDevice
+                        .getProperties2<vk::PhysicalDeviceProperties2,
+                                vk::PhysicalDeviceAccelerationStructurePropertiesKHR>()
+                        .get<vk::PhysicalDeviceAccelerationStructurePropertiesKHR>();
+
+        const Device::RayTracingProperties rayTracingProperties{ rayTracingPipelineProperties.shaderGroupHandleSize,
             rayTracingPipelineProperties.shaderGroupBaseAlignment,
-            accelerationStructureProperties.minAccelerationStructureScratchOffsetAlignment
-        };
+            accelerationStructureProperties.minAccelerationStructureScratchOffsetAlignment };
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Vulkan/Private/Device.cpp:215]: comma

File: Source/Engine/Render/Vulkan/Private/Swapchain.cpp [hunk 15]
```diff
@@ -98,8 +98,7 @@ namespace Details
     static vk::CompositeAlphaFlagBitsKHR SelectCompositeAlpha(vk::SurfaceCapabilitiesKHR capabilities)
     {
         const std::vector<vk::CompositeAlphaFlagBitsKHR> compositeAlphaFlagBits{
-            vk::CompositeAlphaFlagBitsKHR::ePreMultiplied,
-            vk::CompositeAlphaFlagBitsKHR::ePostMultiplied,
+            vk::CompositeAlphaFlagBitsKHR::ePreMultiplied, vk::CompositeAlphaFlagBitsKHR::ePostMultiplied,
             vk::CompositeAlphaFlagBitsKHR::eInherit
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Vulkan/Private/Swapchain.cpp:102]: comma

File: Source/Engine/Render/Vulkan/Shaders/Private/ShaderCompiler.cpp [hunk 16]
```diff
@@ -128,8 +127,7 @@ namespace Details
             .generalSamplerIndexing = true,
             .generalVariableIndexing = true,
             .generalConstantMatrixVectorIndexing = true,
-        }
-    };
+        } };
```
Comment (ExtinctionHD) [inline Source/Engine/Render/Vulkan/Shaders/Private/ShaderCompiler.cpp:130]: comma

File: Source/Engine/Scene/Components/Private/Components.cpp [hunk 17]
```diff
@@ -3,8 +3,7 @@
 #include "Engine/Scene/Scene.hpp"
 
 HierarchyComponent::HierarchyComponent(Scene& scene_, entt::entity self_, entt::entity parent_)
-    : scene(scene_)
-    , self(self_)
+    : scene(scene_), self(self_)
```
Comment (ExtinctionHD) [inline Source/Engine/Scene/Components/Private/Components.cpp]: break before each initialization

File: Source/Engine/Scene/Private/Primitive.cpp [hunk 18]
```diff
@@ -99,15 +96,13 @@ namespace Details
...
 
-const std::vector<VertexInput> Primitive::kVertexInputs{
-    VertexInput{ { vk::Format::eR32G32B32Sfloat }, 0, vk::VertexInputRate::eVertex },
+const std::vector<VertexInput> Primitive::kVertexInputs{ VertexInput{ { vk::Format::eR32G32B32Sfloat }, 0,
+                                                             vk::VertexInputRate::eVertex },
     VertexInput{ { vk::Format::eR32G32B32Sfloat }, 0, vk::VertexInputRate::eVertex },
     VertexInput{ { vk::Format::eR32G32B32Sfloat }, 0, vk::VertexInputRate::eVertex },
-    VertexInput{ { vk::Format::eR32G32Sfloat }, 0, vk::VertexInputRate::eVertex }
-};
+    VertexInput{ { vk::Format::eR32G32Sfloat }, 0, vk::VertexInputRate::eVertex } };
```
Comment (ExtinctionHD) [inline Source/Engine/Scene/Private/Primitive.cpp]: comma

File: Source/Engine/Scene/Private/SceneLoader.cpp [hunk 19]
```diff
@@ -225,15 +224,10 @@ namespace Details
...
-                GetSamplerMipmapMode(sampler.magFilter),
-                GetSamplerAddressMode(sampler.wrapS),
-                VulkanConfig::kMaxAnisotropy,
-                0.0f, std::numeric_limits<float>::max(),
-                false
-            };
+            const SamplerDescription samplerDescription{ GetSamplerFilter(sampler.magFilter),
+                GetSamplerFilter(sampler.minFilter), GetSamplerMipmapMode(sampler.magFilter),
+                GetSamplerAddressMode(sampler.wrapS), VulkanConfig::kMaxAnisotropy, 0.0f,
+                std::numeric_limits<float>::max(), false };
```
Comment (ExtinctionHD) [inline Source/Engine/Scene/Private/SceneLoader.cpp]: comma

File: Source/Utils/Flags.hpp [hunk 20]
```diff
@@ -9,28 +9,17 @@ class Flags
     const static Flags<TBit, TMask> kNone;
     const static Flags<TBit, TMask> kAll;
 
-    constexpr Flags()
-        : mask(0)
-    {}
+    constexpr Flags() : mask(0) {}
```
Comment (ExtinctionHD) [inline Source/Utils/Flags.hpp]: lets try avoid such diff

File: Source/Utils/Assert.hpp [hunk 21]
```diff
@@ -3,7 +3,9 @@
 #include "Utils/Logger.hpp"
 
 #ifdef NDEBUG
-#define Assert(expression) if (!(expression)) std::cout << "Assertion failed: " << #expression << ", file " << __FILE__ << ", line " << __LINE__ << "\n"
+#define Assert(expression)                                                                                             \
+    if (!(expression))                                                                                                 \
+    std::cout << "Assertion failed: " << #expression << ", file " << __FILE__ << ", line " << __LINE__ << "\n"
```
Comment (ExtinctionHD) [inline Source/Utils/Assert.hpp:8]: lets add braces than

**Review by ExtinctionHD — CHANGES_REQUESTED** [review]

File: Source/Utils/Assert.hpp (hunk 21 above)
Comment (mlknz) [reply Source/Utils/Assert.hpp:8]: hm, which braces?

Comment (mlknz) [reply Source/Utils/Assert.hpp:8]: If you propose curled braces **_{}_** - they will screw last semicolon
Now we write **_Assert(x);_** and there is no **_;_** in Assert definition

**Pull request merged**